client/my-sites/theme/main.jsx

```diff
@@ -609,8 +622,12 @@ export default connect(
 		const backPath = getBackPath( state );
 		const currentUserId = getCurrentUserId( state );
 		const isCurrentUserPaid = isUserPaid( state, currentUserId );
-		const theme = getTheme( state, siteIdOrWpcom, id );
+		// Fallback to 'wpcom' source for wpcom themes on Jetpack target sites
+		const theme = getTheme( state, siteIdOrWpcom, id ) || getTheme( state, 'wpcom', id );
 		const error = theme ? false : getThemeRequestErrors( state, id, siteIdOrWpcom );
+		const isWpcomTheme = theme && theme.screenshots;
```

Comment:
We should then also redo the `siteIdOrWpcom` logic inside the theme sheet -- we now have three `ThemeQuery` components, querying:

1. `siteIdOrWpcom` first,
2. then (if Jetpack site) `wpcom`
3. then (if Jetpack site) `wporg`.

We should change that to only 

1. `siteId` (no `OrWpcom`, only if JP)
2. `wpcom`, (unconditionally);
3. `wporg`

Similarly, we might not need the indiscriminate `siteIdOrWpcom` var in other places in the theme sheet anymore but can be more explicit.